### ISSUE TITLE
[DOC] Embedding domain name in search attribute value.

### ISSUE THREAD

**Issue by somindatommy** [wso2/product-is#5111]
Also associates with issue [wso2_product_is#4997](https://github.com/wso2/product-is/issues/4997)

When filtering users using SCIM2, users can can be filtered with populating the domain parameter[1] in request or embedding domain name in the attribute value [2].

However, Embedding domain parameter **will ONLY WORK when the filter operator is either EQ or SW only (WILL NOT WORK for EW and CO operations)**. Therefore, the most preferred method is to request with the domain parameter and count parameter populated in the request. This needs to be documented.

It needs to be documented that **the most recommended method** for filtering is with the **count parameter populated** (Eg: [3]). 

[1] - curl -v -k --user admin:admin 'https://localhost:9443/scim2/Users?filter=userName+ew+1&domain=primary&count=10'
[2] - curl -v -k --user admin:admin 'https://localhost:9443/scim2/Users?filter=userName+ew+primary/1&count=10'

[3] - curl -v -k --user admin:admin 'https://localhost:9443/scim2/Users?filter=userName+ew+1&count=10

Following issues are associated with this issue.
[wos2_product_is#](https://github.com/wso2/product-is/issues/5087)
[wos2_product_is#5082](https://github.com/wso2/product-is/issues/5082)

**Comment by malithie** [comment]
@somindatommy ,
I don't think that we need to support extracting domain with 'ew' conditions at all.
A domain is always prepended to the username. Having the support for 'eq' and 'sw' is enough actually.
'ew' makes no sense in addition, it's more error prone as the respective characters may be a part of the username it self
